sphinx/registry.py

```diff
@@ -61,8 +66,9 @@ def __init__(self):
         self.domain_object_types = {}   # type: Dict[unicode, Dict[unicode, ObjType]]
         self.domain_roles = {}          # type: Dict[unicode, Dict[unicode, Union[RoleFunction, XRefRole]]]  # NOQA
         self.post_transforms = []       # type: List[Type[Transform]]
-        self.source_parsers = {}        # type: Dict[unicode, Parser]
+        self.source_parsers = {}        # type: Dict[unicode, Type[Parser]]
         self.source_inputs = {}         # type: Dict[unicode, Input]
+        self.source_suffix = {}         # type: Dict[unicode, unicode]
         self.translators = {}           # type: Dict[unicode, nodes.NodeVisitor]
         self.transforms = []            # type: List[Type[Transform]]
```

Comment:
From now on, registry class has many container attributes.
They have typehints, but there are no description about structure.

It would be nice if there are descriptions like env module:
```
        # TOC inventory
        self.titles = {}            # type: Dict[unicode, nodes.Node]
                                    # docname -> title node
        self.longtitles = {}        # type: Dict[unicode, nodes.Node]
                                    # docname -> title node; only different if
                                    # set differently with title directive
        self.tocs = {}              # type: Dict[unicode, nodes.Node]
                                    # docname -> table of contents nodetree
        self.toc_num_entries = {}   # type: Dict[unicode, int]
                                    # docname -> number of real entries
```